### PR TITLE
bump SDK to version 1.0.0

### PR DESCRIPTION
**Pull request by joeshaw**
Ticket: https://fastly.atlassian.net/browse/DEVLIB-1135 (edit by @abetts)

### CODE REVIEW

**Review by Integralist — APPROVED** [review]
As long as this SDK release will continue to work for TinyGo, then I'm happy 👍🏻

**Review by dgryski — APPROVED** [review]
LGTM

File: go.sum [hunk 1]
```diff
@@ -1,2 +1,4 @@
 github.com/fastly/compute-sdk-go v0.1.7 h1:DyyneozpIdCn4d/5P9FY0zuMrhEi7ZLV7OwGYtcpgzs=
```
Comment (dgryski) [inline go.sum:1]: Will `go mod tidy` should remove the 0.1.7 lines.

**Review by dgryski — APPROVED** [review]
LGTM (again)

**Comment by joeshaw** [comment]
@Integralist SDK 0.2.0 sets the minimum TinyGo version to 0.28.1.  So it's compatible but I think pulls forward the requirement a bit.

**Comment by Integralist** [comment]
@joeshaw hmm, so if I understand this correctly: merging this PR is going to break the experience for _existing_ CLI users because they will have a CLI (and local config) with constraints set to `0.26.0` as the minimum https://github.com/fastly/cli/blob/main/.fastly/config.toml#L8 and the SDK will not work with that version.

**Comment by joeshaw** [comment]
@Integralist Yeah, if they have an older version of TinyGo and try to use SDK 0.2.0 (including this starter kit post-merge), it'll fail.

**Comment by joeshaw** [comment]
@Integralist The changelog for the 0.2.0 SDK release [mentions that older TinyGo versions are dropped](https://github.com/fastly/compute-sdk-go/blob/main/CHANGELOG.md#changed) and updating the SDK is a conscious decision folks make in their projects -- you don't get upgraded automatically.

So to me it's really an issue of folks using an older CLI (before a 0.28.1 constraint is added) using the starter kit.  Is there a way to specify a minimum required CLI version in the starter kit?  If not, that would be a good mechanism to add for the future but I'm with you that I don't think there's a good way to avoid this otherwise.

**Comment by Integralist** [comment]
> So to me it's really an issue of folks using an older CLI (before a 0.28.1 constraint is added) using the starter kit. 

Yup, exactly.

> Is there a way to specify a minimum required CLI version in the starter kit? 
> If not, that would be a good mechanism to add for the future

Good idea. I'll create a ticket for this.

**Comment by joeshaw** [comment]
@Integralist Are you comfortable with me landing this change now, or do you want to do a CLI release first?

**Comment by Integralist** [comment]
@joeshaw let's get the next CLI release out first as it has some workarounds in place to try and reduce friction for users who upgrade their CLI version but have an existing project: https://github.com/fastly/cli/pull/995

**Comment by joeshaw** [comment]
Marking this as a draft as we'll need to add the build script line, and I'd like to get better confidence that it'll work as expected on Windows.

**Comment by joeshaw** [comment]
`env` appears to work on windows.  See https://github.com/fastly/compute-starter-kit-go-default/actions/runs/5882217274/job/15952358049?pr=8 for more info.  (Those were canceled because I ran `fastly compute serve` -- I wanted to ensure that the output was valid wasm and couldn't think of a better way on Windows.)

**Review by dgryski — APPROVED** [review]
LGTM

**Comment by Integralist** [comment]
@joeshaw 

> env appears to work on windows.

Interesting 🤔 

I'll go back today and retry adding it in and see what happens and let you know.

**Review by Integralist — APPROVED** [review]
LGTM. But just to reiterate we need to wait for new CLI release before merging this.

**Comment by joeshaw** [comment]
Latest commit adds the `scripts.env_vars` field that is going to be added from https://github.com/fastly/cli/pull/995#issuecomment-1684278770.  However, that'll require a new CLI release to work so I've marked this PR as a draft again.

**Comment by Integralist** [comment]
https://github.com/fastly/cli/releases/tag/v10.4.0

**Comment by joeshaw** [comment]
As this will require the latest Fastly CLI release to work, I'm inclined to wait as long as possible to merge this so people will have a change to upgrade.

**Comment by Integralist** [comment]
Makes sense 👍🏻 

Maybe by the time the `1.0.0` of the SDK is ready.

**Pull request merged**